electricitylci/generation_mix.py

```diff
@@ -342,19 +339,150 @@ def olcaschema_genmix(database, gen_dict, subregion=None):
                         break
                 if matching_dict is None:
                     logging.warning(
-                        f"Trouble matching dictionary for {fuelname} - {reg}"
+                        f"Trouble matching dictionary for generation mix {fuelname} - {reg}. Skipping this flow for now"
                     )
                 else:
+                    ra = exchange_table_creation_input_genmix(
+                    database_f1, fuelname
+                    )
+                    ra["quantitativeReference"] = False                    
                     ra["provider"] = {
-                        "name": matching_dict["name"],
-                        "@id": matching_dict["uuid"],
-                        "category": matching_dict["category"].split("/"),
+                       "name": matching_dict["name"],
+                       "@id": matching_dict["uuid"],
+                       "category": matching_dict["category"].split("/"),
                     }
-                exchange(ra, exchanges_list)
-                # Writing final file
+                    #if matching_dict is None:
+                    exchange(ra, exchanges_list)
+                    # Writing final file
 
         final = process_table_creation_genmix(reg, exchanges_list)
+        # print(reg +' Process Created')
+        generation_mix_dict[reg] = final
+    return generation_mix_dict
+
+
+def olcaschema_usaverage(database, gen_dict, subregion=None):
+    if subregion is None:
+        subregion = model_specs.regional_aggregation
+    generation_mix_dict = {}
+
+    us_database = create_generation_mix_process_df_from_egrid_ref_data(subregion='US')
+    #Not choosing the Hawaiian and Alaskan regions. 
+    us_database = us_database[(us_database['Subregion'] != 'HIMS') &  (us_database['Subregion'] != 'HIOA') & (us_database['Subregion'] != 'AKGD') & (us_database['Subregion'] != 'AKMS')]
```

Comment:
And since line 394 includes something similar, it might be best to either add this is parameter (e.g., excluded_regions=["HIMS","HIOA"...] so that we can change this in case we want to use another set of data.

Reply:
I will use this method. Thanks @jump2conclusionsmatt